faq.md

```diff
@@ -1,112 +1,110 @@
 
-Par exemple : 
-```cpp
-std::cout << "Hello world !" << std::endl;
-```
+Précision concernant les bases de données : la section #database concerne principalement les questions relatives aux schémas et aux requêtes. Les questions concernant les bibliothèques spécifiques à leur langages de programmation n'y ont pas leur place.
 
-Qui donnera :
-std::cout << "Hello world !" << std::endl;
+**Sécurité informatique**
 
-/!\ Attention, le caractère ` (accent grave) utilisé ici est différent du ' (guillemet), il peut s'obtenir par la combinaison Alt Gr + 7 sur un clavier azerty.
+Parce que sur ce serveur il y a aussi beaucoup de novices en informatique, nous prenons avec des pincettes tout ce qui touche à la sécurité informatique. Ce sujet n'est abordé *que* lorsqu'il s'agit de renforcer la protection d'un système ou bien de corriger des failles évidentes (exemple: injection sql).
 
-### "J'ai besoin d'aide, venez en MP" est une mauvaise idée (!mp)
-Il est inutile, voire même contre-productif, de demander aux gens de venir en privé pour vous aider. Le serveur dispose d’un grand nombre de canaux sur pratiquement toutes les thématiques du développement, il y en aura forcément un pour votre question.
+Il n'y a personne qui est à la fois habitué, experte reconnue du domaine et qui voudrait devenir ambassadeur d'un canal sécurité. De ce fait nous ne pouvons pas garantir que les discussions concernant le *pentesting*, le *reverse engineering* et d'autres formes de crack/hack se déroule dans un cadre déontologique. Ces discussions ont interdites sur NaN.
 
-De plus, en posant votre question publiquement, l’intelligence collective vous profitera, et vous serez assuré de ne pas recevoir une réponse inadéquate, ou fausse. De ce fait, les demandes d’aides en privé sont interdites sur ce serveur.
+**Web-scrapping**
+
+Le scrapping de données est la pratique de programmatiquement télécharger des données depuis un site. Il est possible de scrapper en forgeant des requêtes HTTP ou bien en automatisant un navigateur web.
+
+Le scrapping n'est pas correctement couvert par un cadre légal. Sur NaN, nous n'autorisons le web-scrapping que lorsque celui-ci peut s'effectuer sans automatisation d'un navigateur (sans selenium) et dans le respect stricte des conditions d'utilisation du site en question. En d'autres termes, lorsqu'un site propose une API et que vous voulez intéragir avec ce site: vous *devez* utiliser cette API.
+
+Toute demande d'aide concernant le scrapping web doit être accompagné d'une référence vers le passage qui autorise le scrapping dans le CGU du site. À défaut, la demande d'aide tombera sous la règle concernant le respect de la loi et sera classée sans suite.
+
+Plus d'informations sur le cadre légal du scrapping web: <https://lexing.be/le-scraping-est-il-legal/>
```

Comment:
```suggestion
**Web-scraping**

Le scraping de données est la pratique de programmatiquement télécharger des données depuis un site. Il est possible de scraper en forgeant des requêtes HTTP ou bien en automatisant un navigateur web.

Le scraping n'est pas correctement couvert par un cadre légal. Sur NaN, nous n'autorisons le web-scraping que lorsque celui-ci peut s'effectuer sans automatisation d'un navigateur (sans Selenium par exemple) et dans le respect stricte des conditions d'utilisation du site en question. En d'autres termes, lorsqu'un site propose une API et que vous voulez intéragir avec ce site : vous *devez* utiliser cette API.

Toute demande d'aide concernant le web-scraping doit être accompagné d'une référence vers le passage qui autorise le scraping dans les CGU du site en question. À défaut, la demande d'aide tombera sous la règle concernant le respect de la loi et sera classée sans suite.

Plus d'informations sur le cadre légal du web-scraping : <https://lexing.be/le-scraping-est-il-legal/>.
```

Reply:
Putain de changement de titre, je le ferai manuellement

Reply:
Je ne suis pas d'accord, le web scraping n'est pas illégal. Certains géants comme Facebook, LinkedIn ont perdu leur procès contre des sites qui faisaient / font du scraping.  

Celui peut être considéré comme illégal dans les CGU que lorsque le bot doit d'abord s'authentifier (par exemple, scraper les profils Facebook de ses amis) 

https://parsers.me/us-court-fully-legalized-website-scraping-and-technically-prohibited-it/

Reply:
De ce fait, je ne pense pas que l'interdire sur NaN soit pertinent. C'est aussi beaucoup moins borderline que certains autres sujets.

Reply:
Le message ne dit pas que c'est illégale, mais que le cadre legal est douteux, de plus tu donne des juriseprudences Americaine, en France tu n'as pas de reglementation (je suppose que c'est la reglementation reference du serveur), je te laisse regarder le lien en fin du paragraphe qui explique les deux points de vue rapidement sur la légalité du data scapping, ce n'est souvent pas le scraping en lui même qui pose problème mais l'utilisation des données derriere

Reply:
Dans #python c'est un problème *très* récurent. Ils y a beaucoup de membres qui viennent poser des questions concernant le scraping de Amazon par exemple, ils veulent automatiser l'achat de cartes graphiques. On a décrété (dans python) qu'on n'aidait *plus jamais* tous ceux qui venaient avec des questions de scraping. Le passage ci-dessous permet de réguler le problème:

> Toute demande d'aide concernant le web-scraping doit être accompagné d'une référence vers le passage qui autorise le scraping dans les CGU du site en question. À défaut, la demande d'aide tombera sous la règle concernant le respect de la loi et sera classée sans suite.

Si on sait nous que le scraping en question ne pose pas de problème (genre qqun a besoin de choper un passage de wikipedia dans le cadre d'un dm) alors on peut tout de même aider. Pour le moment la règle est tacite et tombe en `!borderline` systématiquement. Avec ce passage dans la FAQ on établi notre jurisprudence à ce propos.

D'ailleurs, on ne dit pas que c'est interdit, on dit que c'est compliqué:

> Le scraping n'est pas correctement couvert par un cadre légal.